### ISSUE TITLE
RequireThisCheck: false positive in anonymous class with same name as field in top level class

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#5353]
````
$ cat TestClass.java
public class TestClass {
    int f = 0;

    void method() {
        new AbstractClass() {
            public void methodInner() {
               f = 1; // expecting violation
            }
        };
    }

    public static abstract class AbstractClass {
        int f;

        public abstract void methodInner();
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="RequireThisCheck">
<property name="validateOnlyOverlapping" value="false"/>
</module>
    </module>
</module>

$ java -jar checkstyle-8.8-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:7:16: Reference to instance variable 'f' needs "TestClass.this.". [RequireThis]
Audit done.
Checkstyle ends with 1 errors.
````

I am not expecting a violation on line 7. It is correct that there is a field on line 2 that would need the `this` to access, however, since it is not `final` it cannot be referenced, no matter what, from inside the anonymous class. This check does not verify the anonymous class boundary and requiring `this`.
If you remove the field on line 2, the violation goes away.

Even though a violation here would be correct requiring just `this`, it is not possible to for this check to see class hierarchy in the same file to know this and it wouldn't be able to be extended to multiple files yet. So it is better it produces no violation for now.

**Comment by rnveach** [comment]
Nevermind, code isn't compilable. I will re-examine.

**Comment by rnveach** [comment]
I re-opening this as the example has been fixed to be valid and compilable.

**Comment by Vyom-Yadav** [comment]
I am on it.

**Comment by Vyom-Yadav** [comment]
I am on it.

**Comment by Vyom-Yadav** [comment]
@rnveach Sorry for pinging for such an old issue, just had a doubt that even if the variable `f` is `final` in the `TestClass`, how can it be referenced from inside the anonymous class without making an object of `TestClass` (As both variables are named `f`). Also you are suggesting not show a violation at all in these type of cases, right?

**Comment by Carbon14-48** [comment]
i'm on this !
